File: 프로그래머스/3/152995. 인사고과/인사고과.java
import java.util.*;

class Solution {
    public int solution(int[][] scores) {
        int answer = 0;
        
        int [] wanho = scores[0];
        int wanho_sum = wanho[0] +wanho[1];
        
        Arrays.sort(scores, (a,b) -> { 
            if (a[0] == b[0]) {
                return Integer.compare(a[1],b[1]);
            }
            return Integer.compare(b[0],a[0]);
        });
        
        
        int max_peer  = 0;
        int rank = 1;
        
        for (int [] score : scores)  {
            int att = score[0];
            int peer = score[1];
            
            if (peer < max_peer) {
                if (peer == wanho[1] && att == wanho[0])  {
                    return -1;
                } 
                continue;
            }
            
            max_peer = Math.max(max_peer, peer);
            
            if (wanho_sum < peer +att) {
                rank +=1;
            }  
        }
        return rank;
    }
}

File: 프로그래머스/3/152995. 인사고과/인사고과.py
def solution(scores):
    answer = 0
    
    # 4 5 5 5 3
    
    wanho_sum = scores[0][0] + scores[0][1]
    
    wanho = scores[0] 
    
    scores.sort(key= lambda x : (-x[0],x[1]))
    max_peer = 0 
    rank = 1
    
    
    for attitude, peer in scores :
        if peer < max_peer :
            if attitude == wanho[0] and peer == wanho[1] :
                return -1
            continue 
            
        max_peer = max(max_peer, peer)
        
        if attitude + peer >  wanho_sum :
            rank+=1
            
    return rank